extensions/notification/utils/message.js

```diff
@@ -0,0 +1,79 @@
+const { versioning } = require('arsenal');
+const versionIdUtils = versioning.VersionID;
+
+const constants = require('../constants');
+
+/**
+ * Add attributes to an internal notification message,
+ * with its log entry counterpart.
+ * @param  {Object} logEntry - log entry that generated the message
+ * @param  {Object} message - initial message generated
+ * @return {Object} updated message object
+ */
+function addLogAttributes(logEntry, message) {
+    const { eventMessageProperty } = constants;
+    const msg = {};
+    for (const [key, value] of Object.entries(eventMessageProperty)) {
+        let val = logEntry[value];
+        if (typeof val === 'number') {
+            val = val.toString();
+        }
+        msg[key] = val || null;
+    }
+    return Object.assign(msg, message);
+}
+
+/**
+ * Transforms an event object to specification.
+ * @param  {Object} event - event object
+ * @return {Object} updated event object
+ */
+function transformToSpec(event) {
+    const verId
+        = event.versionId ? versionIdUtils.encode(event.versionId) : null;
+    const message = {
+        Records: [
+            {
+                eventVersion: constants.eventVersion,
+                eventSource: constants.eventSource,
+                awsRegion: event.region,
+                eventTime: event.dateTime,
+                eventName: event.eventType,
+                userIdentity: {
+                    principalId: null,
+                },
+                requestParameters: {
+                    sourceIPAddress: null,
+                },
+                responseElements: {
+                    'x-amz-request-id': null,
+                    'x-amz-id-2': null,
+                },
+                s3: {
+                    s3SchemaVersion: event.schemaVersion,
```

Comment:
Why is the schema version part of the `event` object? I imagined (but can be wrong) that the schema version is a AWS API version like `2006-07-01`

Reply:
I wasn't sure and the doc I referred wasn't clean either, https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html

Reply:
Ok, maybe it's  better to have it defined as a constant `1.0` then, if it refers to the version of this `message` object it should probably not be part of the `event`.

Reply:
@jonathan-gramain there is another property `eventVersion` in the message, this probably is  the one that relates to version of message object. Should I still make the `s3SchemaVersion` to be `1.0`?

Reply:
In doubt, I'd set `s3SchemaVersion` to `1.0` since this is what is shown as an example in the AWS page.